### PR TITLE
Match fox_ka

### PR DESCRIPTION
**Pull request by KiritoDv**
<3

co-authored-by: @sonicdcer
co-authored-by: @petrie911 
co-authored-by: @inspectredc

### CODE REVIEW

File: src/overlays/ovl_i4/fox_ka.c [hunk 1]
```diff
@@ -1,63 +1,2183 @@
+#include "prevent_bss_reordering.h"
```
[the remark this answers is not in the document]
Comment (sonicdcer) [reply src/overlays/ovl_i4/fox_ka.c:1]: Looks like it isn't, can you take it out?

File: src/overlays/ovl_i4/fox_ka.c [hunk 2]
```diff
@@ -1,63 +1,2183 @@
...
+        case 25:
+        case 40:
+            actor = &gActors[5];
+            Matrix_RotateY(gCalcMatrix, actor->obj.rot.y * M_DTOR, 0);
+            Matrix_RotateX(gCalcMatrix, actor->obj.rot.x * M_DTOR, 1);
+            Matrix_MultVec3fNoTranslate(gCalcMatrix, &src, &dest);
+            func_8006EEFC(5, actor->obj.pos.x + (dest.x * 1.5), actor->obj.pos.y + (dest.y * 1.5),
+                          actor->obj.pos.z + (dest.z * 1.5), dest.x, dest.y, dest.z, actor->obj.rot.x, actor->obj.rot.y,
+                          actor->obj.rot.z);
+            break;
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_ka.c:311]: Can the actor temp be removed from this case?

File: src/overlays/ovl_i4/fox_ka.c [hunk 3]
```diff
@@ -1,63 +1,2183 @@
...
+                  (boss->vwork[idx + 1].z * 1.3f) + boss->obj.pos.z, 15.0f);
+    boss->swork[idx + 5] = 60;
+
+    for (i = 0; i < 20; i++) {
+        randX = Rand_ZeroOne();
+        randY = Rand_ZeroOne();
+        randZ = Rand_ZeroOne();
+        func_800794CC((boss->vwork[idx + 1].x * 1.3f) + (boss->obj.pos.x + ((randX - 0.5f) * 300.0f)),
+                      (boss->vwork[idx + 1].y * 1.3f) + (boss->obj.pos.y + ((randY - 0.5f) * 300.0f)),
+                      (boss->vwork[idx + 1].z * 1.3f) + (boss->obj.pos.z + ((randZ - 0.5f) * 300.0f)), 1.55f);
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_ka.c:457]: The Rand temps should be unnecessary.

File: src/overlays/ovl_i4/fox_ka.c [hunk 4]
```diff
@@ -1,63 +1,2183 @@
...
+                Matrix_MultVec3fNoTranslate(gCalcMatrix, &src, &dest);
+                func_800794CC(boss->obj.pos.x + dest.x, boss->obj.pos.y + dest.y, boss->obj.pos.z + dest.z, 1.55f);
+            } else {
+                boss->obj.rot.x -= 0.06f;
+                boss->vel.y = 0.0f;
+                boss->gravity = 0.0f;
+                Math_SmoothStepToF(&boss->unk_078.z, 0.0f, 1.0f, 1.0f, 0.0f);
+                func_i4_80192C08(
+                    (boss->obj.pos.x + 2000.0f) + RAND_FLOAT(500.0f), (boss->obj.pos.y - 500.0f) + RAND_FLOAT(500.0f),
+                    (boss->obj.pos.z + 600.0f) + RAND_FLOAT(1000.0f), 0.0f, 20.0f, 0.0f, RAND_FLOAT(20.0f) + 15.0f);
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_ka.c:1123]:
```suggestion
                    boss->obj.pos.x + 2000.0f + RAND_FLOAT(500.0f), boss->obj.pos.y - 500.0f + RAND_FLOAT(500.0f),
                    boss->obj.pos.z + 600.0f + RAND_FLOAT(1000.0f), 0.0f, 20.0f, 0.0f, RAND_FLOAT(20.0f) + 15.0f);
```

File: src/overlays/ovl_i4/fox_ka.c [hunk 5]
```diff
@@ -1,63 +1,2183 @@
...
+            Math_SmoothStepToF(&boss->fwork[4], boss->fwork[9], 0.05f, 5.0f, 0);
+        }
+        boss->info.hitbox[2] = boss->fwork[0];
+        boss->info.hitbox[12] = boss->fwork[1];
+        boss->info.hitbox[22] = boss->fwork[2];
+        boss->info.hitbox[32] = boss->fwork[3];
+        boss->info.hitbox[43] = ((boss->fwork[4] - 200.0f) + -750.0f);
+        boss->info.hitbox[49] = ((boss->fwork[4] - 200.0f) + -850.0f);
+        boss->info.hitbox[55] = ((boss->fwork[4] - 200.0f) + -950.0f);
+        boss->info.hitbox[61] = ((boss->fwork[4] - 200.0f) + -1200.0f);
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_ka.c:1201]: none of these parentheses are needed

File: src/overlays/ovl_i4/fox_ka.c [hunk 6]
```diff
@@ -1,63 +1,2183 @@
...
+            Matrix_RotateX(gGfxMatrix, -(M_DTOR * 90), 1);
+            Matrix_SetGfxMtx(&gMasterDisp);
+            gSPDisplayList(gMasterDisp++, D_1024AC0);
+            Matrix_Pop(&gGfxMatrix);
+            Matrix_Push(&gGfxMatrix);
+            gDPSetPrimColor(gMasterDisp++, 0, 0, 255, 255, 255, 64);
+            gDPSetEnvColor(gMasterDisp++, 0, 255, 255, 64);
+            Matrix_Translate(gGfxMatrix, 0.0f, 500.0f, 0.0f, 1);
+            Matrix_Scale(gGfxMatrix, boss->fwork[14] * 3.0f, boss->fwork[14] * 3.0f, boss->fwork[14] * 3.0f, 1);
+            Matrix_RotateX(gGfxMatrix, -(M_DTOR * 90), 1);
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_ka.c:1320]:
```suggestion
            Matrix_RotateX(gGfxMatrix, -90 * M_DTOR, 1);
```
I'd prefer to have these in the form `(degree angle) * M_DTOR`

File: src/overlays/ovl_i4/fox_ka.c [hunk 7]
```diff
@@ -1,63 +1,2183 @@
...
+                    break;
+
+                case 550:
+                    D_80177830 = 0;
+                    break;
+
+                case 1010:
+                    Audio_PlaySfx(0x09000002U, &player->sfxPos[0], 0, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                    player->unk_194 = 5.0f;
+                    player->unk_190 = 5.0f;
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_ka.c:1585]:
```suggestion
                    player->unk_190 = player->unk_194 = 5.0f;
```
Does this match?

File: src/overlays/ovl_i4/fox_ka.c [hunk 8]
```diff
@@ -1,63 +1,2183 @@
...
+    }
+    Matrix_RotateY(gCalcMatrix, (player->unk_114 + player->unk_0E8 + 180.0f) * M_DTOR, 0);
+    Matrix_RotateX(gCalcMatrix, -((player->unk_120 + player->unk_0E4) * M_DTOR), 1);
+    src.x = 0.0f;
+    src.y = 0.0f;
+    src.z = player->unk_0D0;
+    Matrix_MultVec3fNoTranslate(gCalcMatrix, &src, &dest);
+    player->vel.x = dest.x;
+    player->vel.z = dest.z;
+    player->vel.y = dest.y;
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_ka.c:1661]: Can these be swapped?

File: src/overlays/ovl_i4/fox_ka.c [hunk 9]
```diff
@@ -1,63 +1,2183 @@
...
+                D_8015F920 = 1;
+                func_8001D444(0, 32797, 0, 255);
+            }
+
+            func_i4_8019848C();
+
+            PRINTF("KT_time %d\n", D_i4_801A0540);
+
+        case 2:
+            D_i4_801A0540 += 1;
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_ka.c:1842]:
```suggestion
            D_i4_801A0540++;
```
Also that PRINTF seems more likely to be here than in the previous case.

File: src/overlays/ovl_i4/fox_ka.c [hunk 10]
```diff
@@ -1,63 +1,2183 @@
...
+                            (func_80031900(actor) != 0) && (gActors->state == 2)) {
+                            actor->iwork[0] = 1;
+                        }
+                    } else {
+                        actor->iwork[4] = 0;
+                    }
+
+                    if ((actor->unk_0E6 > 0) &&
+                        ((gActors[actor->unk_0E6].obj.status == 3) || gActors[actor->unk_0E6].state == 6 ||
+                         gActors[actor->unk_0E6].obj.status == 0)) {
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_ka.c:2016]:
```suggestion
                        ((gActors[actor->unk_0E6].obj.status == 3) || (gActors[actor->unk_0E6].state == 6) ||
                         (gActors[actor->unk_0E6].obj.status == 0))) {
```
also actor statuses have an enum

File: src/overlays/ovl_i4/fox_ka.c [hunk 11]
```diff
@@ -1,63 +1,2183 @@
...
+    ySin = __sinf(actor->obj.rot.y * M_DTOR);
+    yCos = __cosf(actor->obj.rot.y * M_DTOR);
+
+    if (state != 0) {
+        xRand = actor->fwork[4] - actor->obj.pos.x;
+        yRand = actor->fwork[5] - actor->obj.pos.y;
+        zRand = actor->fwork[6] - actor->obj.pos.z;
+        if (!((actor->index + gGameFrameCount) & 7)) {
+            actor->fwork[19] = Math_RadToDeg(Math_Atan2F(xRand, zRand));
+            xAngle = sqrtf(SQ(xRand) + SQ(zRand));
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_ka.c:2065]: same as above.

File: src/overlays/ovl_i4/fox_ka.c [hunk 12]
```diff
@@ -1,63 +1,2183 @@
...
+            zAngle = 360.0f - yAngle;
+        }
+        Math_SmoothStepToAngle(&actor->obj.rot.z, zAngle, 0.1f, 3.0f, 0.01f);
+    }
+    actor->obj.rot.x = -actor->unk_0F4.x;
+    actor->obj.rot.y = actor->unk_0F4.y;
+
+    Math_SmoothStepToF(&actor->fwork[0], actor->fwork[1], 0.2f, 1.0f, 0.1f);
+    Math_SmoothStepToF(&actor->fwork[2], actor->fwork[3], 1.0f, 0.1f, 0.1f);
+    zVel = (actor->fwork[0] + actor->fwork[9]) * xCos;
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_ka.c:2097]: I would comment here about this being xzVel, but I know the variable reuse is required to match. ah, devs.

File: src/overlays/ovl_i4/fox_ka.c [hunk 13]
```diff
@@ -1,63 +1,2183 @@
...
+    }
+
+    if (actor->iwork[0] != 0) {
+        actor->iwork[0] = 0;
+        yVel = (-xSin * 200.0f * 0.5f);
+        xVel = (xCos * 200.0f * 0.5f);
+        zVel = (xCos * 200.0f * 0.5f);
+        func_8006EEFC(actor->unk_0E4, actor->obj.pos.x + ((ySin * xVel) * 1.5f), actor->obj.pos.y + (yVel * 1.5f),
+                      actor->obj.pos.z + ((yCos * zVel) * 1.5f), (ySin * (xCos * 200.0f * 0.5f)),
+                      (-xSin * 200.0f * 0.5f), (yCos * (xCos * 200.0f * 0.5f)), actor->obj.rot.x, actor->obj.rot.y,
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_ka.c:2122]: can the temps not be used here?

File: src/overlays/ovl_i4/fox_ka.c [hunk 14]
```diff
@@ -1,63 +1,2183 @@
...
+            (fabsf(actor->obj.pos.z - gPlayer[0].camEye.z) < 4500.0f)) {
+            actor->iwork[23] = 0;
+        }
+    }
+
+    if ((actor->iwork[8] != 0) && (actor->unk_0E4 < 100)) {
+        angle = __sinf(actor->iwork[8] * 400.0f * M_DTOR) * actor->iwork[8];
+        Matrix_RotateY(gGfxMatrix, M_DTOR * angle, 1);
+        Matrix_RotateX(gGfxMatrix, M_DTOR * angle, 1);
+        Matrix_RotateZ(gGfxMatrix, M_DTOR * angle, 1);
```
Comment (petrie911) [inline src/overlays/ovl_i4/fox_ka.c:2156]:
```suggestion
        Matrix_RotateY(gGfxMatrix, angle * M_DTOR, 1);
        Matrix_RotateX(gGfxMatrix, angle * M_DTOR, 1);
        Matrix_RotateZ(gGfxMatrix, angle * M_DTOR, 1);
```

File: src/overlays/ovl_i4/fox_ka.c (hunk 2 above)
Comment (KiritoDv) [reply src/overlays/ovl_i4/fox_ka.c:311]: Sadly nope

File: src/overlays/ovl_i4/fox_ka.c (hunk 3 above)
Comment (KiritoDv) [reply src/overlays/ovl_i4/fox_ka.c:457]: They are not ):

File: src/overlays/ovl_i4/fox_ka.c (hunk 8 above)
Comment (KiritoDv) [reply src/overlays/ovl_i4/fox_ka.c:1661]: Nope ):

File: src/overlays/ovl_i4/fox_ka.c (hunk 13 above)
Comment (KiritoDv) [reply src/overlays/ovl_i4/fox_ka.c:2122]: Nope

**Review by sonicdcer — APPROVED** [review]

**Review by petrie911 — APPROVED** [review]

**Pull request merged**